workflow/workflow.go

```diff
@@ -358,24 +364,24 @@ func (l *loopState) notifySteps() { //nolint:gocognit
 		}
 
 		// This switch checks to see if it's a node that needs to be run.
-		switch node.Item().Kind {
+		switch nodeItem.Kind {
 		case DAGItemKindStepStage:
-			if node.Item().DataSchema == nil {
+			if nodeItem.DataSchema == nil {
 				// This should only happen if the stage doesn't have any input fields.
 				// This may not even get called. That should be checked.
 				break
 			}
 			// We have a stage we can proceed with. Let's provide it with input.
 			// Tries to match the schema
-			if _, err := node.Item().DataSchema.Unserialize(untypedInputData); err != nil {
-				l.logger.Errorf("Bug: schema evaluation resulted in invalid data for %s (%v)", node.ID(), err)
-				l.recentErrors <- fmt.Errorf("bug: schema evaluation resulted in invalid data for %s (%w)", node.ID(), err)
+			if _, err := nodeItem.DataSchema.Unserialize(untypedInputData); err != nil {
+				l.logger.Errorf("Bug: schema evaluation resulted in invalid data for %s (%v)", nodeID, err)
+				l.recentErrors <- fmt.Errorf("bug: schema evaluation resulted in invalid data for %s (%w)", nodeID, err)
```

Comment:
May be worth it for a future major refactor.

Reply:
This suggestion is kind of the _opposite_ of a major refactor, and it would be best implemented on its own or as part of another change which happens to touch these lines (like this one...).